### PR TITLE
chore: collect test results using junit

### PR DESCRIPTION
**Pull request by pmuir**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,9 @@
 # HTML coverage generated by gocov
 cover.html
 
+# JUnit XML
+*junit.xml
```
Comment (hferentschik) [inline .gitignore]: could we not get all these files (including cover.html) generated somewhere under build? I think it is much nicer to generate all artefacts, be it binaries or test results, under a common base directory. If you want to clean up, you just delete the build directory.

File: Makefile [hunk 2]
```diff
@@ -119,6 +119,11 @@ BUILDFLAGS += -ldflags "-X $(ROOT_PACKAGE)/pkg/features.FeatureFlagToken=$(FEATU
 RELEASE_ORG_REPO := cloudbees/cloudbees-jenkins-x-distro
 endif
 
+ifdef JUNIT_REPORT
+# REPORT_PIPE must go at end of line
+REPORT_PIPE := -test.v 2>&1 | tee /dev/tty | go-junit-report > integration.junit.xml
```
Comment (hferentschik) [inline Makefile]: See above.

File: Makefile [hunk 3]
```diff
@@ -38,6 +38,8 @@ GITHUB_ACCESS_TOKEN := $(shell cat /builder/home/git-token 2> /dev/null)
 FEATURE_FLAG_TOKEN := $(shell cat /builder/home/feature-flag-token 2> /dev/null)
 CGO_ENABLED = 0
 
+REPORTS_DIR=reports/
```
Comment (hferentschik) [inline Makefile]: I'd go for `build/reports`, but having a directory to collect this all is a step forward.

File: jenkins-x-integration.yml [hunk 4]
```diff
@@ -56,10 +58,15 @@ pipelineConfig:
               - name: test-integration
                 #image: docker.io/golang:1.11.5
                 # needs helm in the image for install_gitops_integration_test.go
-                image: gcr.io/jenkinsxio/builder-go:0.1.332
+                image: gcr.io/jenkinsxio/builder-go:0.0.0-SNAPSHOT-PR-457-51
```
Comment (hferentschik) [inline jenkins-x-integration.yml]: A snapshot/pr image?

File: Makefile [hunk 5]
```diff
@@ -119,6 +121,12 @@ BUILDFLAGS += -ldflags "-X $(ROOT_PACKAGE)/pkg/features.FeatureFlagToken=$(FEATU
 RELEASE_ORG_REPO := cloudbees/cloudbees-jenkins-x-distro
 endif
 
+ifdef JUNIT_REPORT
+# REPORT_PIPE must go at end of line
+OUTFILE := $(shell mktemp)
+REPORT_PIPE := -test.v 2>&1 | tee $(OUTFILE) | go-junit-report > build/integration.junit.xml & tail -f $(OUTFILE)
```
Comment (hferentschik) [inline Makefile]: Should this be  $(REPORTS_DIR)/integration.junit.xml or do you see REPORTS_DIR only as target for html report?

Comment (hferentschik) [reply Makefile]: Where `go-junit-report` coming from? Should that not be added to `get-test-deps`?

File: jenkins-x-integration.yml (hunk 4 above)
Comment (pmuir) [reply jenkins-x-integration.yml]: Yes.

File: Makefile (hunk 5 above)
Comment (pmuir) [reply Makefile]: > Should this be $(REPORTS_DIR)/integration.junit.xml or do you see REPORTS_DIR only as target for html report?

Typo. Should be fixed now.

Comment (pmuir) [reply Makefile]: > Where go-junit-report coming from? Should that not be added to get-test-deps?

The builder image. No.

File: Makefile [hunk 6]
```diff
@@ -38,6 +38,8 @@ GITHUB_ACCESS_TOKEN := $(shell cat /builder/home/git-token 2> /dev/null)
 FEATURE_FLAG_TOKEN := $(shell cat /builder/home/feature-flag-token 2> /dev/null)
 CGO_ENABLED = 0
 
+REPORTS_DIR=build/reports/
```
Comment (hferentschik) [inline Makefile]: 👍

File: Makefile [hunk 7]
```diff
@@ -119,6 +121,12 @@ BUILDFLAGS += -ldflags "-X $(ROOT_PACKAGE)/pkg/features.FeatureFlagToken=$(FEATU
 RELEASE_ORG_REPO := cloudbees/cloudbees-jenkins-x-distro
 endif
 
+ifdef JUNIT_REPORT
+# REPORT_PIPE must go at end of line
+OUTFILE := $(shell mktemp)
+REPORT_PIPE := -test.v 2>&1 | tee $(OUTFILE) | go-junit-report > $(REPORTS_DIR)/integration.junit.xml & tail -f $(OUTFILE)
```
Comment (hferentschik) [inline Makefile]: 👍

File: .gitignore [hunk 8]
```diff
@@ -10,8 +10,8 @@
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 
-# HTML coverage generated by gocov
-cover.html
+# Reports
+reports/
```
Comment (hferentschik) [inline .gitignore]: I don't think this would be required anymore

File: Makefile [hunk 9]
```diff
@@ -271,7 +281,7 @@ release-distro:
 
 .PHONY: clean
 clean: ## Clean the generated artifacts
-	rm -rf build release cover.out cover.html
+	rm -rf build release cover.out $(REPORTS_DIR)
```
Comment (hferentschik) [inline Makefile]: REPORTS_DIR is implicitly removed

**Pull request merged**